server/service/osquery.go

```diff
@@ -1412,6 +1412,10 @@ func (svc *Service) preProcessOsqueryResults(ctx context.Context, osqueryResults
 		unmarshaledResults = append(unmarshaledResults, result)
 	}
 
+	if queryReportsDisabled {
+		return unmarshaledResults, nil
```

Comment:
since we can return a nil map here, can we return earlier in saveResultLogsToQueryReports?